### PR TITLE
Ready to merge: Accept x-frame domain value in operand configuration

### PR DESCRIPTION
**Pull request by rashmi43**

### CODE REVIEW

**Comment by rashmi43** [comment]
MCMP is handling it locally but we can decide whether to merge this or not for 3.5.4. If not we can do it for Q4.

**Review by znhafeez — APPROVED** [review]

**Review by yannizhang2019 — APPROVED** [review]

**Comment by rashmi43** [comment]
@posriniv please review

File: pkg/apis/operator/v1alpha1/authentication_types.go [hunk 1]
```diff
@@ -105,6 +105,7 @@ type ConfigSpec struct {
 	FIPSEnabled                 bool   `json:"fipsEnabled"`
 	ROKSEnabled                 bool   `json:"roksEnabled"`
 	NONCEEnabled                bool   `json:"nonceEnabled"`
+	XFrameDomain                string `json:"xframeDomain"`
```
Comment (posriniv) [inline pkg/apis/operator/v1alpha1/authentication_types.go:108]: @rashmi43 you must add `omitempty` here

File: pkg/controller/authentication/ingress.go [hunk 2]
```diff
@@ -758,6 +759,12 @@ func platformOidcBlockIngress(instance *operatorv1alpha1.Authentication, scheme
 func platformOidcIngress(instance *operatorv1alpha1.Authentication, scheme *runtime.Scheme) *net.Ingress {
 
 	reqLogger := log.WithValues("Instance.Namespace", instance.Namespace, "Instance.Name", instance.Name)
+	var xframeDomain string
+	if (instance.Spec.Config.XFrameDomain != "") {
+		xframeDomain = strings.Join([]string{"'ALLOW-FROM ", instance.Spec.Config.XFrameDomain, "'"}, "")
```
Comment (posriniv) [inline pkg/controller/authentication/ingress.go:764]: you can set a flag here here called ingressUpdateRequired and Update the corresponding ingress as we upate the configmap here https://github.com/IBM/ibm-iam-operator/blob/master/pkg/controller/authentication/configmap.go#L173-L178

**Review by posriniv — CHANGES_REQUESTED** [review]

**Pull request merged**